extension/src/popup/ducks/cache.ts

```diff
@@ -27,20 +35,30 @@ type SaveDomainPayload = Record<PublicKey, HomeDomain>;
 
 type SaveTokenLists = AssetListResponse[];
 
+type SaveTokenDetailsPayload = { contractId: string } & TokenDetailsResponse;
+
 interface InitialState {
   balanceData: {
     [network: string]: Record<PublicKey, AccountBalancesInterface>;
   };
   icons: Record<AssetCode, IconUrl>;
   homeDomains: Record<PublicKey, HomeDomain>;
   tokenLists: AssetListResponse[];
+  tokenDetails: {
+    [contractId: string]: TokenDetailsResponse;
+  };
+  historyData: {
+    [network: string]: Record<PublicKey, HistoryResponse>;
+  };
```

Comment:
let's cache tokenDetails and historyData. tokenDetails are unlikely to change, but we can refresh historyData as needed